test/unit/test_store.py

```diff
@@ -0,0 +1,291 @@
+
+logging.getLogger('stevedore.extension').level = logging.INFO
+
+
+class MockLocation(object):
+    def __init__(self, image_id):
+        self.store_location = StoreLocation({'image_id': image_id}, {})
+
+
+class TestStoreLocation(unittest.TestCase):
+    """Tests for scality_glance_store.store.StoreLocation"""
+
+    def test_process_specs(self):
+        image_id = str(uuid.uuid4())
+        store_location = StoreLocation({'image_id': image_id}, {})
+
+        self.assertEqual(image_id, store_location.image_id)
+
+    def test_get_uri(self):
+        image_id = str(uuid.uuid4())
+        store_location = StoreLocation({'image_id': image_id}, {})
+
+        self.assertEqual('scality://%s' % image_id, store_location.get_uri())
+
+    def test_parse_uri(self):
+        image_id_1 = str(uuid.uuid4())
+        store_location = StoreLocation({'image_id': image_id_1}, {})
+
+        image_id_2 = str(uuid.uuid4())
+        store_location.parse_uri('scality://%s' % image_id_2)
+
+        self.assertEqual(image_id_2, store_location.image_id)
+
+
+@mock.patch('eventlet.spawn', mock.Mock())
+class TestStore(glance_store.tests.base.StoreBaseTest):
+    """Tests for scality_glance_store.store.Store"""
+
+    def setUp(self):
+        """Establish a clean test environment."""
+        super(TestStore, self).setUp()
+
+        self.set_sproxyd_endpoints_in_conf(['http://localhost:81/proxy/path/'])
+
+    def set_sproxyd_endpoints_in_conf(self, endpoints):
+        self.conf.set_override('scality_sproxyd_endpoints', endpoints,
+                               group='glance_store')
+
+    def test_init(self):
+        endpoints = set(['http://h1:81/proxy/', 'http://h2:81/proxy/'])
+        self.set_sproxyd_endpoints_in_conf(endpoints)
+
+        store = Store(self.conf)
+        self.assertEqual(endpoints, store.sproxyd_client.sproxyd_urls_set)
+
+    def test_get_schemes(self):
+        store = Store(self.conf)
+
+        self.assertEqual(('scality',), store.get_schemes())
+
+    @mock.patch('scality_sproxyd_client.sproxyd_client.SproxydClient.'
+                'get_object', side_effect=scality_sproxyd_client.exceptions.
+                SproxydException())
+    def test_get_with_sproxyd_exception(self, mock_get_object):
+        store = Store(self.conf)
+
+        image_id = str(uuid.uuid4())
+        location = MockLocation(image_id)
+
+        self.assertRaises(glance_store.exceptions.RemoteServiceUnavailable,
+                          store.get, location)
+        mock_get_object.assert_called_once_with(image_id)
+
+    def test_get(self):
+        store = Store(self.conf)
+
+        image_id = str(uuid.uuid4())
+        location = MockLocation(image_id)
+
+        data = '*'*80
+        headers = {'Content-length': len(data)}
+
+        def gen():
+            yield data
+
+        mock_get_object = mock.Mock()
+        mock_get_object.return_value = headers, gen()
+
+        with mock.patch('scality_sproxyd_client.sproxyd_client.SproxydClient.'
```

Comment:
Move the patch string to the next line if it's too long, the `.` at the end is a bit confusing.